legos/cryptocurrency.py

```diff
@@ -24,12 +37,125 @@ def handle(self, message):
             logger.error('''Could not identify message source in message:
                         {}'''.format(message))
 
+        message_list = message['text'].split()
+        if message_list[0] == '!hodl':
+            self.reply(message, self._parse_multi_commands(message_list), opts)
+        else:
+            try:
+                query = message['text'].split()[1]
+            except:
+                self.reply(message, "Invalid query", opts)
+            self.reply(message, self._lookup_symbol(query), opts)
+
+    def _parse_multi_commands(self, message_list):
```

Comment:
You're falling into the same trap we fell into with the proto-legobot back in 2013-14. This pattern didn't scale well for us back then and probably won't in this case either.

Reply:
'splain this one more plz

Reply:
We tried to write something along a similar pattern where we'd basically split the message and then branch on each option. It ended up getting messy, which was what drove us to the plugin model.

Using argparse for positional arguments or splitting hodl into another lego might be the way to go given how many branches we're starting to accumulate.

Reply:
Resolved by moving hodl to its own lego